### ISSUE TITLE
Publish from ffmpeg

### ISSUE THREAD

**Issue by xmike402** [jchristn/WatsonWebserver#27]
Hello Joel,

I'm trying to use your project to publish from ffmpeg, however I am struggling to accept a raw post.

FFMPEG will simply "send data" to a http address, no intro, no hello nothing just http and ... data.
Example: ffmpeg -i rtsp://camera/ -f mpegts -c:v mpeg1video -s 768x432 -b:v 1670k -bf 0 -c:a mp2 -ar 44100 -ac 1 -b:a 64k -muxdelay 0.001 http://watsonserver/watsonrocks

I can't seem to get the raw data to fly. Ideas or suggestions on how I might just accept the raw data.

In short, I am trying to implement https://github.com/phoboslab/jsmpeg/blob/master/websocket-relay.js with your amazing project.

Thank you!

**Comment by jchristn** [comment]
Hi @xmike402, thanks for your note!  I took a look at the link you sent.  It looks like the implementation you are working on building is a websocket implementation - is that correct?  

If so, one area I'd need to enhance Watson would be to support persistent connections and also handling of upgrade requests and 1xx status messages.

If it's not websockets, and ffmpeg is truly just sending data to an HTTP address with no websocket setup (i.e. using a POST or otherwise) then Watson should absolutely be able to support it.

Do you have another webserver that you have ffmpeg working with currently?  If so, would you be able to put Charles Proxy (or similar) in between ffmpeg and the server to capture what ffmpeg is doing over the wire?

Thanks and have a great day!

**Comment by jchristn** [comment]
Hi @xmike402 if the expectation is that the readers will see all the data from the beginning, writing to a file is probably a good idea.  You may want to consider returning a FileStream in HttpResponse.DataStream (just set the HttpResponse.ContentLength using the FileInfo.Length for the file).  i.e. something like...

```
long contentLength = new FileInfo("/opt/published_data").Length;
FileStream fs = new FileStream("/opt/published_data", FileMode.Open, FileAccess.Read);

return new HttpResponse(request, 200, [headers], [contenttype], contentLength, fs);
```

WatsonWebserver will close the stream for you when done.  Setting ```FileMode.Open``` and ```FileAccess.Read``` will allow multiple concurrent readers.  No FIFO needed.

**Comment by jchristn** [comment]
As of right now responses terminate the connection.  I can add chunked transfer support for the response in a future release but haven't scoped it :)  I'll put it on my list though!

**Comment by xmike402** [comment]
Great to hear. I can't quite suggest where to add such functionality, whether it is handled inside of HttpResponse or if you create a new class. I've been googling around and it does seem the once rarely used chunked encoding is now becoming a major way of communication since more and more the length is unknown and web servers serve more than the web! 
Thanks for your time and help working with to assist.
